int64 horizontal
int64 left
int64 right